pages/algorithms/semantic-role.md

```diff
@@ -0,0 +1,10 @@
+---
+title: Semantic Role 
+key: semantic-role
+---
+
+A semantic association that indicates an object's type. This allows tools to present and support interaction with the object in a manner that is consistent with user expectations about other objects of that type.
```

Comment:
This is more of a description of the role, rather than a definition of it. It doesn't explain how to get the role, for example in cases where a role is invalid, but there is a fallback. I think this needs to take accessibility support into account, including support for fallback roles.

Reply:
Parsing of the role attribute is defined in https://www.w3.org/TR/html/dom.html#aria-role-attribute, a link to which is found in the next paragraph.